### PR TITLE
Refactor: add an output interface class

### LINKED ISSUE #2532: the 'print' function in ESolver_KS_LCAO should be simplified
### Describe Current Status and Possible Solution

if print == true, the code prints out charge density, density matrix, as well as kinetic energy density (?).
I consider the codes should be moved to somewhere else. A new interface should be designed.
The code lines lie from line 746 to line 829.

in ESolver_KS_LCAO::afterscf function:
The same things occur from line 837 to 859 (generally, there should be no 'new' and 'delete' in ESolver)
The same things occur from line 861 to line 949.
The same things occur from line 962 to line 1000 (combine out_pot==1 and out_pot==2 into one interface function).
The same things occur from line 1066 to line 1095 (combine out_mat_hsR, out_mat_t, out_mat_dh, and out_mat_r into one interface function).

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by hongriTianqi**
Fix #2532

### CODE REVIEW

File: source/module_esolver/esolver_ks.h [hunk 1]
```diff
@@ -69,7 +70,13 @@ namespace ModuleESolver
         // "PW/LCAO" ALGORITHM --------------- ION=   1  ELEC=   1--------------------------------
         void writehead(std::ofstream& ofs_running, const int istep, const int iter);
 
-// TODO: control single precision at input files
+        // create a new Output_Rho object
```
Comment (dyzheng) [inline source/module_esolver/esolver_ks.h]: Doxygen format？

File: source/module_esolver/esolver_ks_lcao.cpp [hunk 2]
```diff
@@ -834,53 +762,10 @@ void ESolver_KS_LCAO::afterscf(const int istep)
...
-                    is,
-                    GlobalV::NSPIN,
-                    0,
-                    ssc.str(),
-                    pw_rho->nx,
-                    pw_rho->ny,
-                    pw_rho->nz,
-                    ef_tmp,
-                    &(GlobalC::ucell));
+                this->create_Output_Kin(is, istep).write();
```
Comment (dyzheng) [inline source/module_esolver/esolver_ks_lcao.cpp:749]: Is there existed test case in integrate for this function?

File: source/module_io/output_dm.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#include <string>
+
+#include "module_cell/unitcell.h"
+#include "module_hamilt_lcao/module_gint/grid_technique.h"
+#include "module_io/output_interface.h"
+
+namespace ModuleIO
+{
+
+class Output_DM : public Output_Interface
```
Comment (dyzheng) [inline source/module_io/output_dm.h:14]: Please add some annotations for this class, such as how to use this class.

File: source/module_esolver/esolver_ks_lcao.cpp (hunk 2 above)
Comment (hongriTianqi) [reply source/module_esolver/esolver_ks_lcao.cpp:749]: @dyzheng should I check this with @YuLiu98 or @wenfei-li ?

Comment (hongriTianqi) [reply source/module_esolver/esolver_ks_lcao.cpp:749]: @dyzheng Please check PR #2572

**Review by dyzheng — APPROVED** [review]

**Pull request merged**